builder/processor/src/main/java/io/helidon/builder/processor/GenerateAbstractBuilder.java

```diff
@@ -0,0 +1,1294 @@
+            if (!equalityFields.isEmpty()) {
+                pw.print(" && ");
+            }
+        }
+        if (!hasSuper && equalityFields.isEmpty()) {
+            pw.print("true");
+        } else {
+            pw.print(equalityFields.stream()
+                             .map(field -> {
+                                 if (field.typeName().array()) {
+                                     return "java.util.Arrays.equals(" + field.name() + ", other." + field.getterName() + "())";
+                                 }
+                                 if (field.typeName().primitive()) {
+                                     return field.name() + " == other." + field.getterName() + "()";
+                                 }
+
+                                 return "Objects.equals(" + field.name() + ", other." + field.getterName() + "())";
+                             })
+                             .collect(Collectors.joining(" && ")));
+        }
+        pw.println(";");
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("@Override");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("public int hashCode() {");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        if (equalityFields.isEmpty()) {
+            // no fields on this type
+            if (hasSuper) {
+                pw.println("return super.hashCode();");
+            } else {
+                // hashcode is a constant, as there are no fields and no super type
+                pw.println("return 1;");
+            }
+        } else {
+            if (hasSuper) {
+                pw.print("return 31 * super.hashCode() + Objects.hash(");
+            } else {
+                pw.print("return Objects.hash(");
+            }
+
+            pw.print(equalityFields.stream()
+                             .map(PrototypeProperty::name)
+                             .collect(Collectors.joining(", ")));
+            pw.println(");");
+        }
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+    }
+
+    private static void toString(PrintWriter pw,
+                                 TypeContext typeContext,
+                                 String typeName,
+                                 boolean hasSuper,
+                                 List<CustomMethods.CustomMethod> prototypeMethods,
+                                 boolean isBuilder) {
+        if (prototypeMethods.stream()
+                .map(CustomMethods.CustomMethod::generatedMethod)
+                .map(CustomMethods.GeneratedMethod::method)
+                .filter(it -> "toString".equals(it.name()))
+                .filter(it -> it.returnType().equals(STRING_TYPE))
+                .anyMatch(it -> it.arguments().isEmpty())) {
```

Comment:
how about the redundant/ignore check? where is that done?

Reply:
These lines are used to stop generating `toString()` if there is a custom one defined through custom methods.
Filtering of fields that should be part of toString happens a few lines below, 
`.filter(PrototypeProperty::toStringValue)`